compiler/luci-compute/src/DepthwiseConv2D.cpp

```diff
@@ -0,0 +1,117 @@
+/* Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci_compute/Types.h"
+#include "luci_compute/DepthwiseConv2D.h"
+
+#include "ConvertTypes.h"
+
+#include <tensorflow/lite/kernels/internal/reference/depthwiseconv_float.h>
+
+#include <cassert>
+#include <cstdint>
+
+namespace luci
+{
+namespace compute
+{
+
+bool DepthwiseConv2D::prepare(void)
+{
+  // TODO support other ranks if necessary
+  if (_input_shape.rank() != 4 || _filter_shape.rank() != 4)
+    return false;
+  // if bias exist, check if rank is 1
+  if (_bias_data && _bias_shape.rank() != 1)
+    return false;
+
+  auto const input_batches = _input_shape.dim(0).value();
+  auto const input_height = _input_shape.dim(1).value();
+  auto const input_width = _input_shape.dim(2).value();
+  auto const input_depth = _input_shape.dim(3).value();
+
+  auto const filter_height = _filter_shape.dim(1).value();
+  auto const filter_width = _filter_shape.dim(2).value();
+  auto const filter_channels_out = _filter_shape.dim(3).value();
+
+  if (filter_channels_out % input_depth != 0)
+    return false; // wrong input/output depth ratio
+
+  if (_params.depth_multiplier != static_cast<int32_t>(filter_channels_out / input_depth))
+    return false; // wrong depth multiplier value
+
+  if (_bias_shape.dim(0).value() != filter_channels_out)
+    return false; // unsupported bias value
+
+  // TODO compute output
+  auto output_height = 0;
+  auto output_width = 0;
+
+  _output_shape.rank(4);
+  _output_shape.dim(0) = input_batches;
+  _output_shape.dim(1) = output_height;
+  _output_shape.dim(2) = output_width;
+  _output_shape.dim(3) = filter_channels_out;
+
+  // TODO compute padding
+  _params.padding_values.height = 0;
+  _params.padding_values.width = 0;
```

Comment:
Is it ok to hard-code this?

Reply:
There will be actual computation in next PR.
I've left `// TODO compute padding` comment.
If you are curious about the next change, please look at #12099